### PR TITLE
Return 404 NotFound if requested attachment does not exist

### LINKED ISSUE #20884: GetReleaseAttachment API returns 500 when the ID is not found
### Description

See [here](https://github.com/go-gitea/gitea/blob/6784a707d101ef31a9ef3a46468c28e1d9700ac8/routers/api/v1/repo/release_attachment.go#L58-L62):

```
	attach, err := repo_model.GetAttachmentByID(ctx, attachID)
	if err != nil {
		ctx.Error(http.StatusInternalServerError, "GetAttachmentByID", err)
		return
	}
```

It should return 404 when attachID is not found instead.

### Gitea Version

dev

### Can you reproduce the bug on the Gitea demo site?

Yes

### Log Gist

a

### Screenshots

a

### Git Version

a

### Operating System

a

### How are you running Gitea?

a

### Database

SQLite

### PR DESCRIPTION
**Pull request by zeripath**
Add code to test if GetAttachmentByID returns an ErrAttachmentNotExist error
and return NotFound instead of InternalServerError

Fix #20884

Signed-off-by: Andrew Thornton <art27@cantab.net>

### CODE REVIEW

File: routers/api/v1/repo/release_attachment.go [hunk 1]
```diff
@@ -100,6 +104,10 @@ func ListReleaseAttachments(ctx *context.APIContext) {
 	releaseID := ctx.ParamsInt64(":id")
 	release, err := models.GetReleaseByID(ctx, releaseID)
 	if err != nil {
+		if repo_model.IsErrAttachmentNotExist(err) {
```
Comment (lunny) [inline routers/api/v1/repo/release_attachment.go]: But `GetReleaseID` will never return `ErrAttachmentNotExist`.

**Review by lunny — CHANGES_REQUESTED** [review]

File: routers/api/v1/repo/release_attachment.go (hunk 1 above)
Comment (zeripath) [reply routers/api/v1/repo/release_attachment.go]:
```suggestion
		if models.IsErrReleaseNotExist(err) {
```

File: routers/api/v1/repo/release_attachment.go [hunk 2]
```diff
@@ -166,6 +174,10 @@ func CreateReleaseAttachment(ctx *context.APIContext) {
...
```
Comment (zeripath) [inline routers/api/v1/repo/release_attachment.go]:
```suggestion
		if  models.IsErrReleaseNotExist(err) {
```

File: routers/api/v1/repo/release_attachment.go [hunk 3]
```diff
@@ -166,6 +174,10 @@ func CreateReleaseAttachment(ctx *context.APIContext) {
 	releaseID := ctx.ParamsInt64(":id")
 	release, err := models.GetReleaseByID(ctx, releaseID)
 	if err != nil {
+		if  models.IsErrReleaseNotExist(err) {
```
Comment (zeripath) [inline routers/api/v1/repo/release_attachment.go]:
```suggestion
		if models.IsErrReleaseNotExist(err) {
```

File: routers/api/v1/repo/release_attachment.go (hunk 1 above)
Comment (zeripath) [reply routers/api/v1/repo/release_attachment.go]: Damn... This makes #20891 especially relevant!

**Review by lunny — APPROVED** [review]

**Review by a1012112796 — APPROVED** [review]

**Review by delvh — APPROVED** [review]

**Comment by zeripath** [comment]
make lgtm work

**Pull request merged**